core/common/lru_cache.hpp

```diff
@@ -0,0 +1,211 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_LRUCACHE
+#define KAGOME_LRUCACHE
+
+#include <algorithm>
+#include <cstdint>
+#include <functional>
+#include <memory>
+#include <optional>
+#include <vector>
+
+#include <boost/assert.hpp>
+
+#include "outcome/outcome.hpp"
+
+namespace kagome {
+
+  namespace {
+
+    template <template <bool> class Lockable, bool IsLockable>
+    struct LockGuard {
+      inline LockGuard(const Lockable<IsLockable> &) {}
+    };
+
+    template <template <bool> class Lockable>
```

Comment:
Please consider using SafeObject https://github.com/soramitsu/kagome/blob/master/core/utils/safe_object.hpp

Reply:
This solution selected for ability switch between thread-safe implementation and non-ts.